Only Linux and the BSDs support the RLIMIT_NPROC resource limit.

--- suricata-7.0.6/src/suricata.c
+++ suricata-7.0.6/src/suricata.c
@@ -3019,7 +3019,7 @@ }
 #endif
 
     if (limit_nproc) {
-#if defined(HAVE_SYS_RESOURCE_H)
+#if defined(HAVE_SYS_RESOURCE_H) && defined(RLIMIT_NPROC)
 #ifdef linux
         if (geteuid() == 0) {
             SCLogWarning("setrlimit has no effet when running as root.");
